pkg/porter/uninstall.go

```diff
@@ -32,16 +40,41 @@ func (p *Porter) UninstallBundle(opts UninstallOptions) error {
 		return err
 	}
 
+	var uninstallErrs error
 	fmt.Fprintf(p.Out, "uninstalling %s...\n", opts.Name)
 	err = p.CNAB.Execute(opts.ToActionArgs(deperator))
 	if err != nil {
+		uninstallErrs = multierror.Append(uninstallErrs, err)
+		if strings.Contains(err.Error(), claim.ErrInstallationNotFound.Error()) {
+			return uninstallErrs
+		}
+
 		if len(deperator.deps) > 0 {
-			return errors.Wrapf(err, "failed to uninstall the %s bundle, the remaining dependencies were not uninstalled", opts.Name)
-		} else {
-			return err
+			uninstallErrs = multierror.Append(uninstallErrs,
+				fmt.Errorf("failed to uninstall the %s bundle, the remaining dependencies were not uninstalled", opts.Name))
+		}
+
+		if opts.Delete && !opts.ForceDelete {
+			uninstallErrs = multierror.Append(uninstallErrs, fmt.Errorf("not deleting installation %s as uninstall was not successful; use --force-delete to override", opts.Name))
 		}
+		if !opts.ForceDelete {
+			return uninstallErrs
+		}
+		// else, we swallow uninstallErrs as opts.ForceDelete is true
```

Comment:
I think it would be helpful to no matter what, print out the errors even on --force-delete. We just will have a 0 exit code for --force-delete even when there are errors. That way you know what happened.